node-notes-cli/command-file.js

```diff
@@ -0,0 +1,119 @@
+import * as fs from 'node:fs/promises';
+
+const command = process.argv[2];
+const input1 = process.argv[3];
+const input2 = process.argv[4];
+if (process.argv.length > 5) {
+  console.error('Please enter up to three arguments. A command followed by two optional arguments.');
+  process.exit(1);
+}
+// valid commands read, create, update, delete
+/**
+ * CLI input to read, add, update and delete notes from a JSON file.
+ */
+switch (command) {
+  case 'read':
+    read();
+    break;
+  case 'create':
+    create();
+    break;
+  case 'update':
+    update();
+    break;
+  case 'delete':
+    deleteEntry();
+    break;
+  default:
+    console.error('Improper command given');
+    process.exit(1);
+}
+
+/**
+ * The read function is the first step for all other functions but
+ * if called to only read will log the contents of the notes.
+ * @returns The parsed JSON Object to be used by other functions
+ */
+async function read() {
+  try {
+    const theObject = await fs.readFile('data.json', 'utf8');
+    const parsedObject = JSON.parse(theObject);
+    if (command === 'read') {
+      for (const [key, value] of Object.entries(parsedObject.notes)) {
+        console.log(`${key}: ${value}`);
+      }
+    }
+    return parsedObject;
+  } catch (err) {
+    console.error('An error occured:', err);
+  }
+}
+/**
+ * The create function takes the read JSON file and adds a new entry to notes
+ * and updates the nextID property of dat.json and rewrites the file.
+ * Added bug fix to make sure user inputs a new note when using create command.
+ * @returns returns new data.json text
+ */
+
+async function create() {
+  try {
+    if (!input1) {
+      console.error('This command requires an additional argument to create a note.');
+      process.exit(1);
+    }
+    const objectParsed = await read();
+    const currentID = objectParsed.nextId;
+    objectParsed.notes[currentID] = input1;
+    objectParsed.nextId++;
+    const newFile = await fs.writeFile('data.json', JSON.stringify(objectParsed, null, 2));
```

Comment:
You encapsulated `read`, so why not `write`? Your code would be better if you encapsulate it so you can put all the logic about how to format the notes. For example, think about what would happen if you decided to write the notes to a database instead of the file system? You would need to make multiple changes throughout the file.